### PR TITLE
Account button

### PR DESCRIPTION
**Pull request by keithtkj**
Collaborators: Menelik, Keith, Pakaranjot, Alexander
Business radio button account creation and businessRegister in NetworkingService.swift

### CODE REVIEW

File: Civility Optics.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -13,6 +13,10 @@
 		80AC89FD28032A6E00F68D59 /* profile.swift in Sources */ = {isa = PBXBuildFile; fileRef = 80AC89FC28032A6E00F68D59 /* profile.swift */; };
 		80B5D18028CCF94300379E26 /* FistsView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 80B5D17F28CCF94300379E26 /* FistsView.swift */; };
```
Comment (Spellbreeze) [inline Civility Optics.xcodeproj/project.pbxproj:14]: This file should not be committed.

File: Civility Optics.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 2]
```diff
@@ -1,16 +1,14 @@
 {
-  "object": {
```
Comment (Spellbreeze) [inline Civility Optics.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:2]: This file should not be committed.

File: Civility Optics/ViewModels/BusinessRegisterViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
+//
```
Comment (Spellbreeze) [inline Civility Optics/ViewModels/BusinessRegisterViewModel.swift:1]: This file is unused, don't include it.

File: Civility Optics/ViewModels/BusinessRegisterViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    NetworkingService.businessRegister(email: email, password: password, business_key: business_key, business_name: business_name, business_addr: business_addr) { result in
+      AuthService.current.token = result?.token
+      DispatchQueue.main.async {
+        self.success = true
+      }
+    }
+    print("bRegister ended for ", email)
+  }
+  
+  func taskButton() {
```
Comment (Spellbreeze) [inline Civility Optics/ViewModels/BusinessRegisterViewModel.swift:23]: Remove testing functions before committing.

File: Civility Optics/Views/VenueDetails.swift [hunk 5]
```diff
@@ -77,6 +77,7 @@ var email: String
           HStack {
             VStack {
                 HStack(spacing: 4) {
```
Comment (Spellbreeze) [inline Civility Optics/Views/VenueDetails.swift:79]: Why is this file changed?

File: Civility Optics/Views/Onboarding/BusinessRegisterView.swift [hunk 6]
```diff
@@ -0,0 +1,152 @@
+//
+//  RegisterView.swift
```
Comment (Spellbreeze) [inline Civility Optics/Views/Onboarding/BusinessRegisterView.swift:2]: Is this view used?

**Review by Spellbreeze — APPROVED** [review]

File: Civility Optics.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 7]
```diff
@@ -1,16 +1,14 @@
...
-      {
-        "package": "WrappingHStack",
-        "repositoryURL": "https://github.com/dkk/WrappingHStack",
-        "state": {
-          "branch": null,
-          "revision": "d1fc706a585b9d4d72fb9fe4e5c761cba44d8c0d",
-          "version": "2.0.0"
-        }
+  "pins" : [
+    {
```
Comment (parkaransingh) [inline Civility Optics.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: file is not gonna be commited! Fix first

**Review by keithtkj — COMMENTED** [review]
good addition

File: Civility Optics/Views/Onboarding/BusinessRegisterView.swift [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+          }
+          .padding(.horizontal, 14)
+        }
+        .frame(height: 40)
+      }
+      Spacer()
+      
+        
+      NavigationLink(tag: true, selection: $didCreateAccount) {
+        //SearchView(model: .init())
```
[the remark this answers is not in the document]
Comment (keithtkj) [reply Civility Optics/Views/Onboarding/BusinessRegisterView.swift]: yes, seems like its not doing anything

**Review by parkaransingh — APPROVED** [review]
Approval for testing on develop.

**Review by keithtkj — COMMENTED** [review]
business schema looks good

File: Civility Optics/Views/Onboarding/RegisterView.swift [hunk 9]
```diff
@@ -8,216 +8,332 @@
...
+    @State var business_key = ""
+    @State var business_addr = ""
+    
+    var body: some View {
+        ScrollView() {
+            Text("Use your account to view and submit social-justice ratings of local venues.")
+                .multilineTextAlignment(.leading)
+                .foregroundColor(.pale)
+            
+            VStack(spacing: 2) {
```
Comment (keithtkj) [inline Civility Optics/Views/Onboarding/RegisterView.swift:37]: change this to a spacer instead of an empty v stack

File: Civility Optics/Views/Onboarding/BusinessRegisterView.swift [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+                      }
+          
+      } label: { 
+        EmptyView()
+      }.onChange(of: model.success) { newValue in
+        didCreateAccount = newValue
+      }
+
+      Button { 
+        model.register(email: email, password: password, business_key: business_key,
```
Comment (keithtkj) [inline Civility Optics/Views/Onboarding/BusinessRegisterView.swift]: removing the duplicate business model in the previous commit was a good move

File: Civility Optics/Views/Onboarding/BusinessRegisterView.swift [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+        Text("Business Name")
+          .foregroundColor(.pale)
+          .fontWeight(.semibold)
+        ZStack {
+          RoundedRectangle(cornerRadius: 12)
+            .foregroundColor(.pale)
+          HStack {
+            TextField("Search", text: $business_name)
+              .foregroundColor(.stone)
+              //.keyboardType(.emailAddress)
```
Comment (keithtkj) [inline Civility Optics/Views/Onboarding/BusinessRegisterView.swift]: this can be removed

File: Civility Optics/Views/Onboarding/RegisterView.swift [hunk 12]
```diff
@@ -8,216 +8,332 @@
...
-              Spacer()
+            
+
+       //     if self.accountType == "Reviewer" || self.accountType == "" {
+         //      model = RegisterViewModel()
+           // } else if self.accountType == "Business" {
+             //  model = BusinessRegisterViewModel()
+            //}
+            
+            VStack(spacing: 2) {
```
Comment (keithtkj) [inline Civility Optics/Views/Onboarding/RegisterView.swift:76]: change this to a spacer instead of an empty v stack

File: Civility Optics/Views/Onboarding/BusinessRegisterView.swift [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+        Text("Business Address")
+          .foregroundColor(.pale)
+          .fontWeight(.semibold)
+        ZStack {
+          RoundedRectangle(cornerRadius: 12)
+            .foregroundColor(.pale)
+          HStack {
+            TextField("Search", text: $business_addr)
+              .foregroundColor(.stone)
+              //.keyboardType(.emailAddress)
```
Comment (keithtkj) [inline Civility Optics/Views/Onboarding/BusinessRegisterView.swift]: this is unused and should be removed
